x-pack/plugin/ccr/qa/security/src/test/java/org/elasticsearch/xpack/ccr/FollowIndexSecurityIT.java

```diff
@@ -176,4 +182,51 @@ public void testAutoFollowPatterns() throws Exception {
         pauseFollow(client(), allowedIndex);
     }
 
+    public void testForgetFollower() throws IOException {
+        final String forgetLeader = "forget-leader";
+        final String forgetFollower = "forget-follower";
+        if ("leader".equals(targetCluster)) {
+            logger.info("running against leader cluster");
+            final Settings indexSettings = Settings.builder()
+                    .put("index.number_of_replicas", 0)
+                    .put("index.number_of_shards", 1)
+                    .put("index.soft_deletes.enabled", true)
+                    .build();
+            createIndex(forgetLeader, indexSettings);
+        } else {
+            logger.info("running against follower cluster");
+            followIndex(client(), "leader_cluster", forgetLeader, forgetFollower);
+
+            final Response response = client().performRequest(new Request("GET", "/" + forgetFollower + "/_stats"));
+            final String followerIndexUUID = ObjectPath.createFromResponse(response).evaluate("indices." + forgetFollower + ".uuid");
+
+            assertOK(client().performRequest(new Request("POST", "/" + forgetFollower + "/_ccr/pause_follow")));
+
+            try (RestClient leaderClient = buildLeaderClient(restClientSettings())) {
+                final Request request = new Request("POST", "/" + forgetLeader + "/_ccr/forget_follower");
+                final String requestBody = "{" +
+                        "\"follower_cluster\":\"follow-cluster\"," +
+                        "\"follower_index\":\"" +  forgetFollower + "\"," +
+                        "\"follower_index_uuid\":\"" + followerIndexUUID + "\"," +
+                        "\"leader_remote_cluster\":\"leader_cluster\"" +
+                        "}";
+                request.setJsonEntity(requestBody);
+                logger.info(requestBody);
+                assertOK(leaderClient.performRequest(request));
+
+                final Request retentionLeasesRequest = new Request("GET", "/" + forgetLeader + "/_stats");
+                retentionLeasesRequest.addParameter("level", "shards");
+                final Response retentionLeasesResponse = leaderClient.performRequest(retentionLeasesRequest);
+                final ArrayList<Object> shardsStats =
+                        ObjectPath.createFromResponse(retentionLeasesResponse).evaluate("indices." + forgetLeader + ".shards.0");
+                assertThat(shardsStats, hasSize(1));
+                @SuppressWarnings("unchecked") final Map<String, Object> shardStatsAsMap = (Map<String, Object>)shardsStats.get(0);
+                @SuppressWarnings("unchecked") final Map<String, Object> retentionLeasesStats =
+                        (Map<String, Object>) shardStatsAsMap.get("retention_leases");
+                @SuppressWarnings("unchecked") final ArrayList<Object> leases = (ArrayList<Object>)retentionLeasesStats.get("leases");
```

Comment:
since no unfollow has happened, the shard follow task may add a new lease, right?

Reply:
Not since it is paused.

Reply:
right, I missed that pause follow index api call.